lib/platform.sh

```diff
@@ -2,6 +2,36 @@ function is-released-version {
   [[ ${1} =~ ^([0-9]+\.){0,2}[0-9]$ ]] && return 0
 }
 
+function get-docker-tag {
+  pushd "${DEIS_ROOT}" &> /dev/null
+  echo "git-$(git rev-parse --short HEAD)"
+  popd &> /dev/null
+}
+
+function can-use-ci-artifacts {
+  local docker_tag="$(get-docker-tag)"
+
+  pushd "${DEIS_ROOT}" &> /dev/null
+    if [ -z "$(git status --porcelain)" ]; then
+      # picked store-monitor because it is last to be pushed (alphabetical)
+      docker-hub-contains-image "store-monitor" "${docker_tag}"
+    else
+      rerun_log error "Deis repo ($(pwd)) contains changes, can't use CI artifacts in Docker Hub."
+      return 1
+    fi
+  popd &> /dev/null
+}
+
+function docker-hub-contains-image {
+  image="${1}"
+  docker_tag="${2}"
+  repository="${3:-deisci}"
+
+  curl -s "https://hub.docker.com/v2/repositories/${repository}/${image}/tags/" \
+    | jq '.results[].name' \
```

Comment:
drop it on like `terraform` IMO, tho people may have installed it prior or via `homebrew` so it depends if you want a local version or not :)